src/main/resources/org/cbioportal/persistence/mybatisclickhouse/StudyViewFilterMapper.xml

```diff
@@ -385,58 +384,87 @@
-                    </include>
-                </if>
-                <if test="dataFilterValue.start != null and dataFilterValue.end == null">
-                    AND match(alteration_value, '^>?=?[-+]?[0-9]*[.,]?[0-9]+$')
-                </if>
-                <if test="dataFilterValue.start == null and dataFilterValue.end != null">
-                    AND match(alteration_value, '^&lt;?=?[-+]?[0-9]*[.,]?[0-9]+$')
-                </if>
-                <if test="dataFilterValue.start != null and dataFilterValue.end != null">
-                    AND match(alteration_value, '^[-+]?[0-9]*[.,]?[0-9]+$')
-                </if>
-                <if test="dataFilterValue.start != null or dataFilterValue.end != null">
-                    <choose>
-                        <when test="dataFilterValue.start == dataFilterValue.end">
-                            AND abs(
-                            minus(
-                            <include refid="castStringValueToFloat">
-                                <property name="attribute_value" value="alteration_value"/>
-                            </include>,
-                            ${dataFilterValue.start}
-                            )
-                            ) &lt; exp(-11)
-                        </when>
-                        <otherwise>
-                            <if test="dataFilterValue.start != null">
-                                AND
-                                <include refid="castStringValueToFloat">
-                                    <property name="attribute_value" value="alteration_value"/>
-                                </include> &gt; ${dataFilterValue.start}
-                            </if>
-                            <if test="dataFilterValue.end != null">
-                                AND
+    <sql id="numericalGenomicDataFilter">
+        <!-- check if 'NA' is selected -->
+        <bind name="containsNA" value="false" />
+        <bind name="containsNumericalValue" value="false" />
+        <foreach item="dataFilterValue" collection="genomicDataFilter.values">
+            <choose>
+                <when test="dataFilterValue.value == 'NA'">
+                    <bind name="containsNA" value="true" />
+                </when>
+                <otherwise>
+                    <bind name="containsNumericalValue" value="true" />
+                </otherwise>
+            </choose>
+        </foreach>
+        <if test="containsNA">
+            SELECT DISTINCT sd.sample_unique_id
+            FROM sample_derived sd
+                LEFT JOIN (<include refid="selectAllGenomicNumericalSamples"/>) AS sthiswrongagain ON sd.sample_unique_id = sthiswrongagain.sample_unique_id
+            WHERE alteration_value IS null
+        </if>
+        <if test="containsNA and containsNumericalValue">
+            UNION ALL
+        </if>
+        <if test="containsNumericalValue">
+            SELECT DISTINCT sd.sample_unique_id
```

Comment:
@ryan why do we select from sample_derived here and then JOIN on it.   doesn't seem like we use anything from sd?